### ISSUE TITLE
Switch back to original "GitHub Flavored Markdown"

### ISSUE THREAD

**Issue by howardroark** [isaacs/github#940]
Recently GitHub went ahead with this... https://githubengineering.com/a-formal-spec-for-github-markdown/

This is probably an opinion more than anything else.  I think Markdown is something that should be loose and evolve to stay human friendly rather than be tied to a hardened specification driven by mainly programmers.   I know the originator of Markdown on many occasions stated that he does not approve of this direction. 

The GitHub Markdown parser used to be ok with things like spaces in urls, as are web browsers.  This is not the case any longer.  I guess I just struggle to understand why something as simple as markdown needs a complex specification.  Lots of people use Markdown that are not programmers and GitHub should encourage this I would think.

This should be enough of a spec... https://daringfireball.net/projects/markdown/

**Comment by rmcardle** [comment]
A formal specification is like a contract between the users and the developers. It lets the users know what they can expect from the software and gives the developers a clear target work towards. That doesn't mean it can't change and evolve, but there needs to be some place common to both groups for these expectations to be documented.

Without a formal specification, the source code becomes the specification. Users shouldn't be expected to have to read the source code to determine the expected behavior of the software. Developers shouldn't have to guess at how users will expect their software to work.

You may disagree with parts (or even all) of the specification, but don't just shoot down the whole idea of a formal specification because of that.

If you think spaces should be allowed in URLs, petition to change that part of the specification. Don't try and get the entire specification dropped.

**Comment by howardroark** [comment]
Yeah that's fair!  I guess I just feel that a rigid specification ends up making it less approachable by curious newcomers because it tends harden up too much.  In the beginning someone looked at John's article and made a common sense attempt to implement it into GitHub and it worked quite nicely I felt. A light airy contract that focused on the idea that Markdown is meant to be simple and easy.

I think it's a bit too late to get things to change in the existing spec.  If only it had a higher level philosophy to guide it.... Like "Be approachable to anyone looking to write more efficiently."

**Comment by bcoe** [comment]
speaking as someone who's worked a bit on npm's markdown implementation (for package READMEs) it's nice to have a spec to work from. This helps us make sure that the README looks almost identical on npmjs.com, as it looks on github.com.

**Comment by howardroark** [comment]
Fair!  Though it's a very developer focused point.  I just agree with John Gruber that the point of Markdown is to be less about developers and more about writing.  Markdown should be evolving in how it attempts to better enable any person to write efficiently.  In my opinion the current spec setup is hyper focused on developers and also pretty much got locked down within a few weeks of starting up. 

I just feel that CommonMark was less about liking Markdown and more about not liking that it  didn't have an "official spec". But yeah I get it :S
